wait/host_port.go

```diff
@@ -207,10 +215,16 @@ func internalCheck(ctx context.Context, internalPort nat.Port, target StrategyTa
 			return fmt.Errorf("%w, host port waiting failed", err)
 		}
 
+		/*
+			Docker maps exit code 127 to 126, while Podman treats them separately.
+			Handle both to ensure compatibility with Docker and Podman.
+		*/
 		if exitCode == 0 {
 			break
-		} else if exitCode == 126 {
+		} else if exitCode == exitEaccess {
 			return errShellNotExecutable
+		} else if exitCode == exitCmdNotFound {
+			return errShellNotFound
```

Comment:
This would also need to be handled by callers, so we'll need additional updates.